src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,319 +76,294 @@ const defaultProps = {
+                                    return;
+                                }
+
+                                if (messageEditInput === relatedTargetId) {
+                                    return;
+                                }
+                                openReportActionComposeViewWhenClosingMessageEdit();
+                            }}
+                            selection={selection}
+                            onSelectionChange={(e) => setSelection(e.nativeEvent.selection)}
+                        />
+                    </View>
+                    <View style={styles.editChatItemEmojiWrapper}>
+                        <EmojiPickerButton
+                            isDisabled={props.shouldDisableEmojiPicker}
+                            onModalHide={() => InteractionManager.runAfterInteractions(() => textInputRef.current.focus())}
+                            onEmojiSelected={addEmojiToTextBox}
+                            nativeID={emojiButtonID}
+                        />
+                    </View>
+
+                    <View style={styles.alignSelfEnd}>
+                        <Tooltip text={props.translate('common.saveChanges')}>
+                            <PressableWithFeedback
+                                style={[styles.chatItemSubmitButton, hasExceededMaxCommentLength ? {} : styles.buttonSuccess]}
+                                onPress={publishDraft}
+                                nativeID={saveButtonID}
+                                disabled={hasExceededMaxCommentLength}
+                                accessibilityRole="button"
+                                accessibilityLabel={props.translate('common.saveChanges')}
+                                hoverDimmingValue={1}
+                                pressDimmingValue={0.2}
+                            >
+                                <Icon
+                                    src={Expensicons.Checkmark}
+                                    fill={hasExceededMaxCommentLength ? themeColors.icon : themeColors.textLight}
+                                />
+                            </PressableWithFeedback>
+                        </Tooltip>
                     </View>
                 </View>
-                <ExceededCommentLength
-                    comment={this.state.draft}
-                    onExceededMaxCommentLength={this.setExceededMaxCommentLength}
-                />
-            </>
-        );
-    }
+            </View>
+            <ExceededCommentLength
+                comment={draft}
+                onExceededMaxCommentLength={(hasExceeded) => setHasExceededMaxCommentLength(hasExceeded)}
+            />
+        </>
+    );
 }
 
 ReportActionItemMessageEdit.propTypes = propTypes;
 ReportActionItemMessageEdit.defaultProps = defaultProps;
+ReportActionItemMessageEdit.displayName = 'ReportActionItemMessageEdit';
 export default compose(
     withLocalize,
```

Comment:
It was reverted for an unrelated reason, but let's re-introduce the `useLocalize` hook from https://github.com/Expensify/App/pull/20708 and use it here instead of `withLocalize`.